x/stakeibc/keeper/host_zone.go

```diff
@@ -189,6 +189,36 @@ func (k Keeper) IncrementValidatorSlashQueryProgress(
 	return nil
 }
 
+// Increments the number of validator delegation changes in progress by 1
+// Note: This modifies the original host zone struct. The calling function must Set this host zone
```

Comment:
Why would it be bad to set the host zone inside this function so we don't have this footgun calling convention?

Reply:
This goes back to that thing I mentioned earlier about where a lot of this is triggered from hooks where we don't revert state if an error is thrown, so it's good to push the set operations as far downstream as possible

We'll come back and fix this once we add the state revert code, and once we add the validators as a stand alone prefix store